source/extensions/common/wasm/context.cc

```diff
@@ -1589,7 +1589,14 @@ void Context::setEncoderFilterCallbacks(Envoy::Http::StreamEncoderFilterCallback
   encoder_callbacks_ = &callbacks;
 }
 
-void Context::onHttpCallSuccess(uint32_t token, Envoy::Http::ResponseMessagePtr& response) {
+void Context::onHttpCallSuccess(uint32_t token, Envoy::Http::ResponseMessagePtr&& response) {
+  if (proxy_wasm::current_context_ != nullptr) {
+    // We are in a reentrant call, so defer.
+    auto response_ptr = response.release();
+    wasm()->addAfterVmCallAction([this, token, response_ptr] {
```

Comment:
Nit: wouldn't this be more readable?

```
wasm()->addAfterVmCallAction([this, token, response = response.release()] {
```

to make it clearer that we're moving this value to lambda?

Reply:
google3 went from 11->17 directly so I wasn't sure if that was a 14 thing or not.  Nice to know that it is.